components/overflow-group/overflow-group.js

```diff
@@ -428,48 +466,20 @@ class OverflowGroup extends RtlMixin(LocalizeCoreElement(LitElement)) {
 		return items.filter(({ isHidden }) => !isHidden);
 	}
 
-	_getItems() {
+	async _getItems() {
 		// get the items from the button slot
-		this._slotItems = this._getSlotItems();
+		this._slotItems = await this._getSlotItems();
 		// convert them to layout items (calculate widths)
 		this._itemLayouts = this._getItemLayouts(this._slotItems);
 		// convert to dropdown items (for overflow menu)
-		this._dropdownItems = this._slotItems.map((node) => convertToDropdownItem(node));
-	}
-
-	_getOverflowMenu() {
-		if (this._overflowMenuHidden) {
-			return;
-		}
-		const moreActionsText = this.localize('components.overflow-group.moreActions');
-		const overflowItems = this._dropdownItems ? this._dropdownItems.slice(this._chompIndex) : [];
-		const menu = html`<d2l-dropdown-menu>
-			<d2l-menu label="${moreActionsText}">
-				${overflowItems}
-			</d2l-menu>
-		</d2l-dropdown-menu>`;
-
-		if (this._mini) {
-			return html`<d2l-dropdown-more class="d2l-overflow-dropdown-mini" text="${moreActionsText}">
-				${menu}
-			</d2l-dropdown-more>`;
-		}
-
-		if (this.openerStyle === OPENER_STYLE.SUBTLE) {
-			return html`<d2l-dropdown-button-subtle class="d2l-overflow-dropdown" text="${moreActionsText}">
-				${menu}
-			</d2l-dropdown-button-subtle>`;
-		}
-
-		return html`<d2l-dropdown-button class="d2l-overflow-dropdown" text="${moreActionsText}">
-			${menu}
-		</d2l-dropdown-button>`;
+		this._dropdownItems = this._slotItems.map((node) => convertToOverflowItem(node));
 	}
 
-	_getSlotItems() {
-		const nodes = this._buttonSlot.assignedNodes({ flatten: true });
-		const filteredNodes = nodes.filter((node) => {
-			const isNode = node.nodeType === Node.ELEMENT_NODE && node.tagName.toLowerCase() !== 'template';
+	async _getSlotItems() {
+		const filteredNodes = await filterAsync(this._buttonSlot.assignedNodes({ flatten: true }), async(node) => {
```

Comment:
This follows the same pattern used with `tag-list` (https://github.com/BrightspaceUI/core/blob/main/components/tag-list/tag-list.js#L311) in order to make sure a node is ready before we size it.